# Maddalena's Notebook
## Open Science 2022/2023

## 23/03/23 

I started exploring the course material and the data sources we will need to use according to our research question

<b>Research Question</b>
* What is the coverage of publications in Social Science and Humanities (SSH) journals 
(according to ERIH-PLUS) included in OpenCitations Meta? 
* What are the disciplines that have more publications? 
* What are countries providing the largest number of publications and journals? 
* How many of the SSH journals are available in Open Access according to the data in DOAJ?

<b>data sources</b>
* <a href= "http://opencitations.net/meta">OpenCitations Meta</a>
* <a href= "https://kanalregister.hkdir.no/publiseringskanaler/erihplus/">ERIH-PLUS</a>
* <a href= "https://doaj.org/">DOAJ</a>

One of the first problems to address in our research will be how to retrieve the data, whether through REST API or by a dump download, hence I tried to figure out how a REST API works and then I tried making some requests to OpenCitations REST API.
Since we will need to access a large amount of data but at the moment I see no way of doing it manually, I am wondering if there are way of doing it by accessing the REST API with Python and I plan to look into it.


I downloaded ERIH-PLUS list of approved journals as a dump and I tried to explore the dataset

In [12]:
import pandas as pd

approvedSSH = pd.read_csv("F:\DATAvarie\ERIHPLUSapprovedJournals.csv", encoding = "utf-8", sep=";")
print(len(approvedSSH))
approvedSSH.info()
approvedSSH

11065
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 11065 entries, 0 to 11064
Data columns (total 9 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   Journal ID              11065 non-null  int64 
 1   Print ISSN              8741 non-null   object
 2   Online ISSN             9510 non-null   object
 3   Original Title          11065 non-null  object
 4   International Title     11065 non-null  object
 5   Country of Publication  10945 non-null  object
 6   ERIH PLUS Disciplines   11065 non-null  object
 7   OECD Classifications    11065 non-null  object
 8   [Last Updated]          11065 non-null  object
dtypes: int64(1), object(8)
memory usage: 778.1+ KB


Journal ID Print ISSN Online ISSN  \
0          486254  1989-3477         NaN   
1          488561        NaN   2341-0515   
2          504135        NaN   2068-3472   
3          495209  2250-4591   2346-9986   
4          488332        NaN   0719-5737   
...           ...        ...         ...   
11060      486933  1995-4328   1995-4336   
11061      496777  2082-4939         NaN   
11062      502441  1409-939X   1857-968X   
11063      496467        NaN   2535-0390   
11064      491147  2309-1606         NaN   

                                          Original Title  \
0                     @tic.revista d'innovació educativa   
1      [i2] Investigación e Innovación en Arquitectur...   
2                                        [Inter]sections   
3                 +E: Revista de Extensión Universitaria   
4                                                 100-Cs   
...                                                  ...   
11060  ΣΧΟΛΗ (Schole). Philosophskoe antikovedenie i ...   
11061              Wrocławsko-Lwowskie Zeszyty Prawnicze   
11062                 ЕтноАнтропоЗум / EthnoAnthropoZoom   
11063                              Русистика без граници   
11064          Філософія освіти. Philosophy of Education   

                                     International Title  \
0                     @tic.revista d'innovació educativa   
1      [i2] Investigación e Innovación en Arquitectur...   
2                                        [Inter]sections   
3                 +E: Revista de Extensión Universitaria   
4                                                 100-Cs   
...                                                  ...   
11060  ΣΧΟΛΗ (Schole). Ancient philosophy and the cla...   
11061           Вроцлавсько-львівський юридичний збірник   
11062                 ЕтноАнтропоЗум / EthnoAnthropoZoom   
11063  Русистика без границ; Russian studies without ...   
11064          Філософія освіти. Philosophy of Education   

      Country of Publication  \
0                      Spain   
1                      Spain   
2                    Romania   
3                  Argentina   
4                      Chile   
...                      ...   
11060                 Russia   
11061                 Poland   
11062              Macedonia   
11063               Bulgaria   
11064                Ukraine   

                                   ERIH PLUS Disciplines  \
0      Interdisciplinary research in the Social Scien...   
1      Art and Art History, Cultural Studies, Human G...   
2      Gender Studies, Cultural Studies, Literature, ...   
3      Interdisciplinary research in the Humanities, ...   
4      Interdisciplinary research in the Social Sciences   
...                                                  ...   
11060  Archaeology, Art and Art History, Classical St...   
11061  History, Business and Management, Economics, L...   
11062  Anthropology, Cultural Studies, Religious Stud...   
11063  History, Cultural Studies, Linguistics, Litera...   
11064     Pedagogical & Educational Research, Philosophy   

                                    OECD Classifications       [Last Updated]  
0            Educational Sciences; Other Social Sciences  2015-06-25 13:48:26  
1      Arts (Arts, History of Arts, Performing Arts, ...  2016-04-18 17:34:55  
2      Languages and Literature; Other Humanities; So...  2022-10-18 08:40:28  
3                Other Humanities; Other Social Sciences  2023-02-02 17:14:12  
4                                  Other Social Sciences  2018-04-26 16:33:26  
...                                                  ...                  ...  
11060  Arts (Arts, History of Arts, Performing Arts, ...  2016-02-09 09:58:08  
11061  Economics and Business; History and Archaeolog...  2020-09-28 15:44:29  
11062  Languages and Literature; Philosophy, Ethics a...  2022-04-07 09:17:03  
11063  Educational Sciences; History and Archaeology;...  2019-04-03 10:04:29  
11064  Educational Sciences; Philosophy, Ethics an

Nonetheless these informations are about journals and not about publication. I am thinking we will need a list of DOI of the publications indexed in ERIH-PLUS, I have explored a bit but I can't seem to find articles' downloadable data. This will be something to address in the first group meeting.


## 27/03/23 

I read Emerald publishing guidelines on how to write an abstract. 

According to the key points that an abstract should feature, I revised the sources provided by the teacher on our research question subject and did some further research to better focus what is the purpose and value of our research.

In particular, since the domain of our research question is Scientometrics and inquires about the coverage of SSH journal in OpenCitatins and how many of those are open access, I noted down some articles I skimmed and plan to read in the future:

- <a href="https://www.nature.com/articles/s41599-018-0149-x">The possibility and desirability of replication in the humanities</a>
- <a href="https://scholarlypublications.universiteitleiden.nl/handle/1887/65315">Exploration of reproducibility issues in scientometric research</a>
- <a href="https://direct.mit.edu/qss/article/3/4/953/113634/A-quantitative-and-qualitative-open-citation">A quantitative and qualitative open citation analysis of retracted articles in the humanities</a>
- <a href="https://www.degruyter.com/document/doi/10.1515/opis-2018-0013/html">Scholarly Communication Practices in Humanities and Social Sciences: A Study of Researchers' Attitudes and Awareness of Open Access</a>

Finally, I started reasoning on the first point of the abstract "Purpose" by formulating some on point questions to share with the group later today, as well as my personal answer, to direct us for writing our abstract. 

* what is the problem we need to solve? why is it relevant? What is the main finding?
<center>The problem to solve is to understand the state of the art of Social Science and Humanities journals citation data presence in OpenCitation databases. <br> It is relevant because of citations centrality in ssh domain and the possibility of conducting further research on available data on citations. How does Open access relate to that?<br>
The main finding will be 1.providing an up do date overview on citation data and oppennes in the ssh domain. 2. creating a software that can make these data available and keep them up to date for enhancing further research on the topic</center>

<hr>

### slef-notes
* How many of the Open Access journals adopt Article processing charge (= make people pay a fee to publish open access?). What is that for? Where else could the open access system sustain itself without having authors pay to publish?

<b>Notes from <a href="https://www.emeraldgrouppublishing.com/how-to/authoring-editing-reviewing/write-article-abstract">Emerald group publishing guidelines</a></b>

maximum of 250 words

<b>Points to feature</b>
* <b>Purpose</b>: This is where you explain why you undertook this study. explain the problem that you have solved.  let readers know why you chose to study this topic or problem and its relevance. Let them know what your key argument or main finding is.
* <b>Study design, methodology, approach</b>: Let readers know exactly what you did to reach your results.  Used tools, methods, protocols or datasets
* <b>Findings</b>: what you found during your study, whether it answers the problem you set out to explore, and whether your hypothesis was confirmed. Clear, give exact figures and not generalizations.
* <b>Originality, value</b>: analysis of the value of your results. Ask colleagues whether your analysis is balanced and fair. Conjecture what future research steps could be. 

<b>Include</b>
* Reasearch limitations/implications
* Practical implications
* Social implications


After the group meeting - where we shared our current understanding of the research to carry and also envisioned possible study designs and various technical approaches - I offered to write a coherent first draft of the abstract considering the material and perspectives we shared. This first draft is pushed in the repository for my teammates to read and rework on, so we can agree on a final version to present on Wednesday.


## Abstract first draft
<span style="color: red">currently around 400 words, to cut</span>

* <b>Purpose</b>: In the domain of Social Science and Humanities, citations hold a central role in measuring reliability and interconnectedness of a publication as well as its long term impact. Referencing works that inspired a particular work provide the reader with the possibility of accessing primary sources, hence, enhancing scholarly communication and community evaluation of a research.
This project aims to assess the state of the art of Social Science and Humanities journals citation data coverage in OpenCitation databases and investigate how many of these are also Open Access. The research provides further insight into the available data, evaluating which countries and disciplines provide the larger number of publications and journals.

* <b>Study design and methodology</b>: We use data from multiple sources: OpenCitations Meta, a database that stores and delivers bibliographic metadata about citations; ERIH-PLUS, an academic Social Science and humanities journal index; DOAJ, the directory of open access journals to retrieve data about open access journals. These data are fetched, filtered, and processed by means of Python Programming Language. The code will be published and available in a GitHub repository, along with instructions on how to use it, to provide users full access to reproduce this analysis at any moment in time. Finally, output data will be visualized in an intuitive and user friendly way.

* <b>Findings</b>: Considering the diverse problematics and the reluctance that revolve around opening the scholarly knowledge in the SSH domain, we expect to find an increasing but still not enough full open access journals. Nonetheless, we expect this deficiency to be in part counterbalanced by a significant number of open citation metadata of SSH publications, provided by OpenCitations  database. <span style="color: red">to inquire</span>

* <b>Originality</b>: Our study provides valuable insights into the current state of the openness of Social Science and Humanities publications and provide reusable data for enhancing further research about Open Science in the Scientometrics domain.
Furthermore, we believe that increased open access to SSH research can enable broader societal engagement, foster cross-disciplinary collaboration, and promote the dissemination of knowledge to a wider audience.

* <b>Limitations</b>: The research is based on the current state and reliability of publicly available data, which might not encompass the entire landscape of SSH publications. These data may and will probably change over time or might come with errors we overlook, despite our best efforts to be as precise an meticulous as possible.

Keywords: Open Access, Social Science and Humanities, OpenCitations Meta, ERIH-PLUS, DOAJ, scholarly communication, open science, citation

## 02/04/23 

Today I revised 29/03 lesson about FAIR and open data and summarized/organized my handwritten notes + slides in a unique txt document.
While revising I noted down some questions and doubts to share and discuss with the group during our next meeting, where we will start working on our Data Management Plan.
<hr>
<b>Unclear issues</b>

- What are practical examples of vocabularies that use FAIR principles? (Interoperability principle 2)
- (Meta)data include qualified references to other (meta)data (?)
About the DMP
- What data (beside csv datasets) will our research produce?
- Do we need to add metadata to our data?
- what metadata should a software have, are there standards?
- Datasets to describe in our DMP are source datasets or our final processed dataset?

## 04/04/23 

In order to fill up the sections of the DMP I am in charge of, I first read a <a hred="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1004525">Ten simple rules for creating a good data management PLAN</a> and wrote a small summary, highlighting the parts that may concern us the most.

Then I started writing the DMP first dataset description.
1. data summary 
2. reused data
3. fair data

During the compilation of the Dataset description I stopped at some questions, since I realized I was not entirely sure how to answer them:

<b>1.1.3 What are the formats of the described generated/collected data?</b><br>
I am not sure if a dataset description should be specific for one kind of data or if it can include all kind of data produced during the research? I think --> our generated data will be diffused in different ways, a dataset in the classical term for sure, but then also probably a textual description analysis and likely some visualiztions (maybe a platform? A website? a pdf?)

I looked into some other resources to help me better understand the specific of different data formats
* <a href="https://ukdataservice.ac.uk/learning-hub/research-data-management/format-your-data/recommended-formats/">File formats recommended by the UK Data Service</a>
* <a href="http://www.docs.is.ed.ac.uk/docs/data-library/EUDL_RDM_Handbook.pdf">Edinburgh University Data Library Research Data Management Handbook</a>
* <a href="https://dmptool-stg.cdlib.org/general_guidance">DMPTool</a>

In the dropdown menu I chose "Discipline specific formats" since it seemed to me the most suitable to our type of data. All the formats we will use are in common usage by the research community, even though not "discipline specific" in a narrow sense.

<b>2.1.1 Are you re-using the described data and how?</b><br>
since we are using public structured data I initially though this meant we were reusing data. Anyway, on a second thought, since they seem to intend "reused data" as data that are outcomes of other researches, this is not true for us: our source data is downloaded from public directories, aggregators and databases.

<b>3.1.1.1 Will you use metadata to describe the data?</b>
I am familiar with DCAT standard but I though it could be worth looking into other standards amongst the ones present in the dropdown menu. I had a quick look at <a href="https://www.w3.org/TR/vocab-data-cube/#outline">The RDF data cube vocabulary</a> and at <a href="https://schema.datacite.org/meta/kernel-4.4/doc/DataCite-MetadataKernel_v4.4.pdf">The DataCite metadata schema</a>.
Reading the RDF data cube vocabulary, made me think about whether we will be also producing statystical data and in what format they need to be published (I think csv is fine, but also could be visualizations) and with which standard they should be described (DCAT is also fine but we could look into SDMX)<br>
<b>N.B.</b> DCAT has the class dcat:Resource which is a general catalogue resource: "It is strongly recommended to use a more specific sub-class. When describing a resource which is not a dcat:Dataset or dcat:DataService, it is recommended to create a suitable sub-class of dcat:Resource, or use dcat:Resource with the dct:type property to indicate the specific type." this would allow as to include non-datasets resurces as data in our catalogue too.

<b>3.1.1.12 Will you provide searchable metadata for the described data?</b>
How can we make our metadata searchable?

As it turns out, many other answers require further group discussion. So far, considering these questions led me to note down useful resources to envision possible answers.

<hr>

* <a href="https://www.openaire.eu/find-trustworthy-data-repository">data repositories</a>
* <a href="http://www.openarchives.org/OAI/2.0/guidelines.htm">OAIP-MH</a>





## 05/04/23 

I filled up the FAIR section of the software dataset in our DMP. For applying FAIR principles to software data, this resource was very helpful: <a href="https://fair-software.nl/">Five reccommendation for fair software</a>

Later, Sebastiano and I had a meeting to revise together every part of the DMP and discuss the parts we individually filled.

## 07/04/23 

I had a look at Ali's additions in the DMP than <a href="https://zenodo.org/record/7809055#.ZDBTZ3ZBxEY">uploaded it</a> on Zenodo 
The first version of our DMP has the following doi:10.5281/zenodo.7809055 

Then, I revised 30/03 lesson about FAIR and open data and summarized/organized my handwritten notes + slides in a unique txt document.

This is a draft of a methodology for a preliminary Literature Review (Introduction to a primary research topic) for our research.
Criteria: 
* key words: OoenCitations, Open Access, SSH Journals
* not before year 2020
* I will read the selected paper's abstract and check how detailed is their methodology section (if present) or if they are related to an independent methodology publications. Then I will select among those, three papers to read more thoroughly.
* Studies providing an independent methodology publication will be preferred.

Finally, I exploerd a bit CWL and protocols.io, running <a href="https://www.protocols.io/view/a-methodology-for-gathering-and-annotating-the-raw-5qpvo5pmbl4o/v2">this</a> workflow + downloaded OpenCitations Meta data dump


## 08/04/23 
I did some further exploration of our datasets in order to better understand how to structure our workflow and share possible ideas or problems during this evening's group meeting.

In [13]:
# exploring ERIH PLUS disciplines for answering research question:
# what are the disciplines that have more publications ?
allcombination = set(approvedSSH['ERIH PLUS Disciplines'])
allcombination = list(allcombination)
disciplines = set()
for el in allcombination:
    if ',' in el:
        disc = (el.split(', '))
        for d in disc:
            disciplines.add(d)
    else: 
        disciplines.add(el)
print(len(disciplines)) #30
print(disciplines)

#OECD classification is another disciplinary classification so probably we will need to map those
allOECDcombination = set(approvedSSH['OECD Classifications'])
allOECDcombination = list(allOECDcombination)
disciplinesOECD = set()
for e in allOECDcombination:
    if ';' in e:
        dis = (e.split('; '))
        for d in dis:
            disciplinesOECD.add(d)
    else: 
        disciplinesOECD.add(e)
print(len(disciplinesOECD)) #14
print(disciplinesOECD)



30
{'History & Philosophy of Science', 'Psychology', 'Anthropology', 'Social Statistics and Informatics', 'Art and Art History', 'Linguistics', 'Business and Management', 'Interdisciplinary research in the Social Sciences', 'Archaeology', 'Philosophy', 'History', 'Gender Studies', 'Sociology', 'Demography', 'Pedagogical & Educational Research', 'Law', 'Human Geography and Urban Studies', 'Science and Technology Studies', 'Religious Studies and Theology', 'Musicology', 'Literature', 'Library and Information Science', 'Cultural Studies', 'Political Sciences and International Relations', 'Interdisciplinary research in the Humanities', 'Economics', 'Environmental Studies', 'Media Studies and Communication', 'Film and Theatre Studies', 'Classical Studies'}
14
{'Social and Economic Geography', 'Media and Communications', 'Languages and Literature', 'Law', 'Psychology', 'Economics and Business', 'Other Humanities', 'Educational Sciences', 'Philosophy, Ethics and Religion', 'History and Archae

<a href="http://help.prod-incites.com/inCites2Live/filterValuesGroup/researchAreaSchema/oecdCategoryScheme/oecd.html">OECD classification</a> is another disciplines classification

As far as ISSN/ESBN are concerned, some of them might have null values as we've seen:

0.   Journal ID : 11065 non-null  int64 
1.   Print ISSN : 8741 non-null   object
2.   Online ISSN : 9510 non-null   object

same goes for Countries:

3. Country of Publication : 10945 non-null (out of 11065 journals)

We need to make sure that at least on of the field "Print ISSN" / "Online ISSN" is non-null because this is the unique identifier that will allow us to connect with OpenCitation meta dataset

In [14]:
#let's check if all 11065 journals are unique through ERIH PLUS unique ID
journals = set(approvedSSH['Journal ID'])
len(journals) #11065
#check how many NaN values in ISSN
nanISSN = approvedSSH['Print ISSN'].isna().sum()
print(nanISSN)
nanISSNOnline = approvedSSH['Online ISSN'].isna().sum()
print(nanISSNOnline)
#are there cases when both of them are null?
noISSN = 0
for idx, row in approvedSSH.iterrows():
    if pd.isna(row['Print ISSN']) and pd.isna(row['Online ISSN']):
        noISSN += 1
print(noISSN) #0 
#pare di no

#also check how many countries are na
noCountry = approvedSSH['Country of Publication'].isna().sum()
print(noCountry) #oioioi 120 


2324
1555
0
120


missing country values might be a problem to answer our research question "What are countries providing the largest number of publications and journals?" but we might be able to retreive them for other datasets.

Now let's explore DOAJ

In [15]:
doaj = pd.read_csv("F:\DATAvarie\journalcsv__doaj_20230404_1935_utf8.csv", encoding="UTF-8") 
doaj.head()
doaj.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 19172 entries, 0 to 19171
Data columns (total 54 columns):
 #   Column                                                                       Non-Null Count  Dtype  
---  ------                                                                       --------------  -----  
 0   Journal title                                                                19172 non-null  object 
 1   Journal URL                                                                  19172 non-null  object 
 2   URL in DOAJ                                                                  19172 non-null  object 
 3   When did the journal start to publish all content using an open license?     19171 non-null  float64
 4   Alternative title                                                            7458 non-null   object 
 5   Journal ISSN (print version)                                                 11088 non-null  object 
 6   Journal EISSN (online version)        

In [16]:
#let's select columns that interest 
doaj = doaj[["Journal title", "Journal ISSN (print version)", "Journal EISSN (online version)", "Keywords", "Publisher", "Country of publisher", "Persistent article identifiers", "Does the journal comply to DOAJ's definition of open access?", "Subjects"]]
doaj.head()

Journal title Journal ISSN (print version)  \
0  Anais da Academia Brasileira de Ciências                    0001-3765   
1                                      ACME                    0001-494X   
2                  Acta Biochimica Polonica                    0001-527X   
3                Acta Dermato-Venereologica                    0001-5555   
4                 Acta Médica Costarricense                    0001-6012   

  Journal EISSN (online version)  \
0                      1678-2690   
1                      2282-0035   
2                      1734-154X   
3                      1651-2057   
4                      2215-5856   

                                            Keywords  \
0  biological sciences, exact and earth sciences,...   
1  italian literature, classic literature, lingui...   
2  molecular biology, biophysics, bioinformatics,...   
3  sexually transmitted infections, psoriasis, ps...   
4  medicine, public health, medical sciences, health   

                                      Publisher Country of publisher  \
0               Academia Brasileira de Ciências               Brazil   
1              Università degli Studi di Milano                Italy   
2                    Polish Biochemical Society               Poland   
3                       Medical Journals Sweden               Sweden   
4  Colegio de Médicos y Cirujanos de Costa Rica           Costa Rica   

  Persistent article identifiers  \
0                            DOI   
1                       DOI, NBN   
2                            DOI   
3                            DOI   
4                            NaN   

  Does the journal comply to DOAJ's definition of open access?  \
0                                                Yes             
1                                                Yes             
2                                                Yes             
3                                                Yes             
4                                                Yes             

                                            Subjects  
0                                            Science  
1                                      General Works  
2  Science: Chemistry: Organic chemistry: Biochem...  
3                              Medicine: Dermatology  
4                                           Medicine

Let's try to merge ERIH-PLUS and DOAJ data on the ISSN, first in print, then online, to keep only SSH Journals that are also in DOAJ directory

In [17]:
oa_sshPrint = pd.merge(doaj, approvedSSH, left_on="Journal ISSN (print version)", right_on="Print ISSN", how="inner")
oa_sshPrint.head() #18789539 which seems a lot, what is repeating?
# we can observe that "country of publisher" and "country of publication" have the same values
oa_sshOL = pd.merge(doaj, approvedSSH, left_on="Journal EISSN (online version)", right_on="Online ISSN", how="inner")
oa_sshOL.head() #1501144 

Journal title Journal ISSN (print version)  \
0                   Acta Psychologica                    0001-6918   
1                           Afro-Ásia                    0002-0591   
2  Anali Pravnog Fakulteta u Beogradu                    0003-2565   
3                      Análise Social                    0003-2573   
4                  Angle Orthodontist                    0003-3219   

  Journal EISSN (online version)  \
0                      1873-6297   
1                      1981-1411   
2                      2406-2693   
3                      2182-2999   
4                            NaN   

                                            Keywords  \
0  clinical psychology, health psychology, cognit...   
1  african studies, diaspora studies, race relati...   
2  international law, law, civil law, criminal la...   
3  sociology, social conditions, political scienc...   
4                odontology, dentistry, orthodontics   

                                           Publisher Country of publisher  \
0                                           Elsevier          Netherlands   
1  Universidade Federal da Bahia - Centro de Estu...               Brazil   
2  University of Belgrade, Faculty of Law, Belgra...               Serbia   
3  Instituto de Ciências Sociais da Universidade ...             Portugal   
4        E.H Angle Education and Research Foundation        United States   

  Persistent article identifiers  \
0                            DOI   
1                            DOI   
2                            DOI   
3                            NaN   
4                            DOI   

  Does the journal comply to DOAJ's definition of open access?  \
0                                                Yes             
1                                                Yes             
2                                                Yes             
3                                                Yes             
4                                                Yes             

                                            Subjects  Journal ID Print ISSN  \
0       Philosophy. Psychology. Religion: Psychology      340169  0001-6918   
1  History (General) and history of Europe: Histo...      503556  0002-0591   
2                                                Law      501706  0003-2565   
3  Political science | Social Sciences: Sociology...      487446  0003-2573   
4                                Medicine: Dentistry      486254  1989-3477   

  Online ISSN                      Original Title  \
0   1873-6297                   Acta Psychologica   
1   1981-1411                           Afro-Ásia   
2   2406-2693  Anali Pravnog fakulteta u Beogradu   
3   2182-2999                      Análise Social   
4         NaN  @tic.revista d'innovació educativa   

                        International Title Country of Publication  \
0                         Acta Psychologica            Netherlands   
1                                 Afro-Asia                 Brazil   
2  Annals of the Faculty of Law in Belgrade                 Serbia   
3                            Análise Social               Portugal   
4        @tic.revista d'innovació educativa                  Spain   

                               ERIH PLUS Disciplines  \
0                                         Psychology   
1  Anthropology, History, Cultural Studies, Socio...   
2  History, History & Philosophy of Science, Econ...   
3  Sociology, Anthropology, History, Political Sc...   
4  Interdisciplinary research in the Social Scien...   

                                OECD Classifications       [Last Updated]  
0                                         Psychology  2023-02-02 17:14:12  
1  History and Archaeology; Languages and Literat...  2022-04-24 14:23:07  
2  Economics and Business; History and Archaeolog...  2021-05-19 11:21:25  
3  History and Archaeology; Political Science; So...  2016-04-10 12:16:35  
4        Educational Sciences; Other Soc

We will probably need to reorganize journal identifiers to actually get all possible matches<br>
Now check if ISSN are repeated in our two dataframes:

In [18]:
uniqueISSN = set(approvedSSH['Print ISSN'])
print(len(uniqueISSN)) #8742

uniqueISSN1 = set(doaj['Journal ISSN (print version)'])
print(len(uniqueISSN1)) #11089

#they both have 8742 unique ISSN sounds wierd
approvedSSH.describe(include="all")
doaj.describe(include="all")

8742
11089


Journal title Journal ISSN (print version)  \
count                   19172                        11088   
unique                  19037                        11088   
top     Revista de Psicología                    0001-3765   
freq                        5                            1   

       Journal EISSN (online version)   Keywords Publisher  \
count                           18209      19172     19171   
unique                          18209      18077      8719   
top                         1678-2690  education  Elsevier   
freq                                1         49       643   

       Country of publisher Persistent article identifiers  \
count                 19172                          14359   
unique                  131                             45   
top               Indonesia                            DOI   
freq                   2235                          14081   

       Does the journal comply to DOAJ's definition of open access?  Subjects  
count                                               19157               19172  
unique                                                  1                3695  
top                                                   Yes            Medicine  
freq                                                19157                 515

I run out of memory so I explored OpenCitation Meta dump manually
<table>
<tr>
<th>id</th>
<th>title</th>
<th>author</th>
<th>issue</th>
<th>volume</th>
<th>venue</th>
<th>page</th>
<th>pub_date</th>
<th>type</th>
<th>publisher</th>
<th>editor</th>
</tr>
<tr>
<td>meta:br/060209 doi:10.4230/lipics.approx/random.2020.19</td>
<td>Distributed Testing Of Graph Isomorphism In The CONGEST Model</td>
<td>Levi, Reut [meta:ra/0610110096 orcid:0000-0003-3167-1766]; Medina, Moti [meta:ra/0612046435 orcid:0000-0002-5572-3754]</td>
<td></td>
<td></td>
<td>[meta:br/060182 issn:1868-8969]</td>
<td></td>
<td>2020</td>
<td>report</td>
<td>Schloss Dagstuhl - Leibniz-Zentrum Für Informatik [meta:ra/0605251]</td>
<td>Byrka, Jarosław [meta:ra/069044096 orcid:0000-0002-3387-0913]; Raghu Meka [meta:ra/0605252]</td>
</tr>
</table>


As we can see, meta has its own internal identifiers (<a href="https://opencitations.net/meta">OMID</a>), <q>The entities subject to deduplication and associated with an OMID are identifiers (abbr. id), agent roles (i.e., authors, editors, publishers, abbr. ar), responsible agents (i.e., people and organisations, abbr. ra), resource embodiments (i.e., pages, abbr. re), venues, volumes, and issues (which are all bibliographic resources, abbr. br).</q><br>
OMIDS areseparated from ISSN and other identifiers by a blank space.<br>

Later, we had the group meeting where we started compiling a workflow in the protocols.io platform.

## 12/04/23 

I completed a first draft of the workflow adding the steps that were missing after our group meeting (steps 1.2, 1.3, 2, 3, 4, 5). I left the vizualization part to be completed by my teammates, as well as any other changes or feedback they may have on the workflow.

This first draft is saved in a pdf: https://drive.google.com/file/d/1MSGXQWTWEzxym7zgHx1CcS2-nI60Nux-/view?usp=sharing 


## 16/04/23 

I revised 12/04 lesson material and summarized it in a doc.
Then, I read Pika.py's protocol and I wrote the review, published in qeios <a href="https://doi.org/10.32388/0FZCPH">https://doi.org/10.32388/0FZCPH</a>

## 22/04/23

I read the other group's Peer reviews of our DMP and protocol and noted down the main things to improve in order to discuss them at our next group meeting

## 27/04/23

I revised and summarized Open Source Software lesson. <br>
Then I proceeded to connect our code repository to Zenodo consulting <a href="https://docs.github.com/en/repositories/archiving-a-github-repository/referencing-and-citing-content#issuing-a-persistent-identifier-for-your-repository-with-zenodo">this page</a>.
I also had a look at how to <a href="https://docs.github.com/en/repositories/releasing-projects-on-github/managing-releases-in-a-repository">manage releases in a repository</a>.
Finally, I did some further research on licences and read <a href="https://www.gnu.org/licenses/license-recommendations.html">this article</a> with license recommendations.

Later I had a look at last year's group material and explored some python libraries: pickle and vaex.

## 02/05/23

We had a group meeting to change the DMP according to the peer reviews received and we adjusted the rebuttal letter Ali wrote to include our specific changes.
Then we published the new dmp on Zenodo obtaining a new  <a href="https://doi.org/10.5281/zenodo.7887693">doi</a> 

## 03/05/23

We had another group meeting to revise the protocol according to the peer reviews we have received and I wrote the rebuttal letters.
Before issuing a new version of the protocol we planned to meet again to discusse further changes we want to include.

Later, I revised and summarised the last two lessons of the course.

## 07/05/23 - 08/05/23

I took some time to study Ali's code in order to be able to write some unittests for it. But I didn't have the time to thoroughly understand it, and I have some doubts and concerns about how issn are matched.

We had a group meeting to revise Ali's code, since it was taking to much time to process the whole data dump and found a way to bring the processing time down to 20 minutes. 
We planned to meet tomorrow to think a bit better about the overall software architecture before coding and also divide the work better.

Later, I studied again the code and run it making some trials with a sample batch of files and formulated some more specific doubts to share tomorrow with the group:

*  by concatenating Print and Online ISSN in the ERIH-PLUS dataset to match them with OC Meta dataset (line 12 of <code>process.py</code>) we are losing the information of which identifiers belong to the same venue. Might this be a problem later when we are merging this result list of matching issn again with ERIH-PLUS dataset? AND then again online and print ones are concatenated (line 29). 

* In general I think we need to be more careful while performing these operations and try the code with files that provide different scenarions. So far I was running with the first 5 csv in the dump and in 4 of them there were no matches of issn, so I didn't manage to test properly some part of the code where dataframes of different batches are concatenated, for example.

* Are we sure that OpenCitations Meta dump always only have 1 issn in the venue column? what happens if there are more? (the <code>.extract()</code> method only matches the first occurrence I understood (line 6 of <code>process.py</code>))

* I think line 40: <code> merged_data['EP_ISSN'] = merged_data['EP_Print_ISSN'].combine_first(merged_data['EP_Online_ISSN'])</code> could be risky since it is creating a new ISSN column by substituting nan values in Print ISSN column with the value in Online ISSN. Both Print and Online ISSN columns are dropped after. This means that we lose the Online ISSN if the Print ISSN was there, keeping just one ISSN for a venue that had multiple issn. However, this might not be a problem since OpenCitations Meta issn are matched before this operation and we might not need to keep multiple ids once we know which internal identifiers are matching. We need to discuss this together with the group. I personally think that keeping all identifiers is a much cleaner and safer way to go.

## 09/05/23

We had a group meeting and discussed doubts. One of the things was how to make easily runnable the software since currently we have file paths inside the functions.

We agreed how to rewrite the workflow and I wrote it while the others were trying to implement a better way to pass file paths as paramenters.

We published on zenodo the new workflow and we are waiting for the new doi to be minted so we can also publish the rebuttal letters.

## 12/05/23

I wrote the first draft of the code to answer research question 2 and 3. Even though it still is very unrefined and I am aware there are several issues I need to fix and double check.

Some self-memos:
- I am retreiving all ERIH-PLUS venues corresponding to each discipline and THEN looking at what venues are in our dataset "resultDf" of OC Meta journals that are also in ERIH-PLUS = not all the venues that I have in the dictionary might be present in resultDF, since they might not be covered by OC Meta, so it does make sense that the number of venue in the dictionary might not correspond to the number of venues in resultDf
- I used EP_id instead of omid because it was actually more practical
- I skipped the part where we try to see if a missing country is in DOAJ
- Some venues have multiple countries 
- still missing the part where the counts are sorted in descending order

The flowchart I produced for extracting disciplines and map them to the venues can be also found in our code repo <code>countries_disc.drawio.png</code>

Here is the code:

In [ ]:
#answer to question 2 and 3 of research: 
#2. What are the disciplines that have more publications?
#3. What are disciplines providing the largest number of publications and journals? 
import pandas as pd
from pprint import pprint

result_df = pd.read_csv("Workflow-Steps-1.1-1.2/resultDf.csv")
erih_df = pd.read_csv("Workflow-Steps-1.1-1.2/ERIHPLUSapprovedJournals.csv", sep= ";")

# step 2.2 and 2.3 of workflow
def create_dict(disc_dict, countr_dict, df):
    empty_countries = []
    for idx, row in df.iterrows():
        
        #disciplines
        if len(disc_dict) == 0:
            disciplines = row["ERIH PLUS Disciplines"].split(', ')
            disc_dict= {key: [row["Journal ID"]] for key in disciplines}
        
        else: 
            disciplines = set(row["ERIH PLUS Disciplines"].split(', '))
            keys = set(disc_dict.keys())
            diff = disciplines - keys

            if len(diff) == 0:
                for key in disciplines:
                    disc_dict[key].append(row["Journal ID"])
            else:
                disc_dict.update({key: [] for key in diff})            
                for key in disciplines:
                    disc_dict[key].append(row["Journal ID"])
            
        #countries. NEED TO UPDATE BECAUSE THERE ARE MULTIPLE COUNTRIES SOMETIMES
        if pd.isna(row["Country of Publication"]):
            empty_countries.append(row["Journal ID"])
            # WRITE A FUNCTION TO SEE IF EMPTY_COUNTRIES CAN BE FOUND IN DOAJ AND CALL IT 
            # for the moment I am saving in a list
            
        else:
            if row["Country of Publication"] not in countr_dict:
                countr_dict[row["Country of Publication"]] = [row["Journal ID"]]
            else: 
                countr_dict[row["Country of Publication"]].append(row["Journal ID"])
        # WRITE A FUNCTION TO SEE IF EMPTY_COUNTRIES CAN BE FOUND IN DOAJ AND CALL IT
                
    return disc_dict, countr_dict, empty_countries 



disc_dict, country_dict, empty_countries = create_dict({}, {}, erih_df)

#print(len(countr_dict)) #114
#print(len(empty_countries)) #121

# step 3 and 4 of workflow actually condensed
def counts(result_df, country_dict, disc_dict):
    result_df = result_df[["EP_id", "Publications_in_venue"]]
    disciplines_count = pd.DataFrame(columns=['Discipline','Journal_count','Publication_count'])
    countries_count = pd.DataFrame(columns=['Country','Journal_count','Publication_count'])

    for key, value in disc_dict.items():
        # venue_count = len(value) # 2125 venues for Interdisciplinary research in the Social Sciences

        venue_in_OCMeta_count =  result_df[result_df["EP_id"].isin(value)]
        venue_in_OCMeta_count.reset_index(drop=True, inplace=True)
        venue_per_disc = len(venue_in_OCMeta_count) # 1592
        pub_per_disc = venue_in_OCMeta_count['Publications_in_venue'].sum() #753265
        
        #create df row
        disc_row = pd.DataFrame([{'Discipline' : key ,'Journal_count' : venue_per_disc , 'Publication_count' : pub_per_disc}])
        disciplines_count = pd.concat([disciplines_count, disc_row], ignore_index = True)
    
    for key, value in country_dict.items():
        venue_in_OCMeta_count =  result_df[result_df["EP_id"].isin(value)]
        venue_in_OCMeta_count.reset_index(drop=True, inplace=True)
        venue_per_country = len(venue_in_OCMeta_count) 
        pub_per_country = venue_in_OCMeta_count['Publications_in_venue'].sum() 
        
        #create df row
        country_row = pd.DataFrame([{'Country' : key ,'Journal_count' : venue_per_country , 'Publication_count' : pub_per_country}])
        countries_count = pd.concat([countries_count, country_row], ignore_index = True)
    
    return disciplines_count, countries_count

''' for el in pub_count["EP_id"].tolist():
            value.remove(el)
        print(value)   
        print(len(value))''' #here I wanted to double check that the number of unmatched value ep_id
#was complementare to the number of the matched one but it is telling me at one point that "el is not in values".
#this shouldn't be possible, given that the df should be a subsection of value list!
#something is up... it happens with EP_id 498742


disciplines_count, countries_count = counts(result_df, country_dict, disc_dict)
#pprint(disciplines_count)
#print("___________________________________________________")
#pprint(countries_count)

disciplines_count.to_csv('Workflow-Steps-2.1-2.2-3-4\disciplines_count.csv', index=False)
countries_count.to_csv('Workflow-Steps-2.1-2.2-3-4\countries_count.csv', index=False)


## 13/05 - 18/05

Saturday 13/05 Seba and I met and we had a look at Ali's code for retrieving Countries and disciplines. At first it was getting different results than mine, but after minor adjustments we managed to get the same outcomes. 
Then we started writing the code to retrieve missing countries from DOAJ dataset.

These days we were discussing if it could be reasonable to organise the code in classes and why it could be preferable than procedural programming approach so I tried to do some research on that. At first I thought it was better to adopt an object oriented approach, enabling code reuse, modularity, and abstraction. But since at this point major parts of the software are already implemented without agreeing on the overall structure, I find it very difficult to encapsulate code parts to create objects that make sense.
Anyway I made an attempt in doing it and my results can be found in our <a href="https://github.com/open-sci/2022-2023-playarists-code/tree/main/oop-approach/remix">repo</a> (link need ot be changed)

* <code>OCMeta_coverage_classes.py</code> for the coverage of OC Meta SSH journals
* <code>Disciplines_Countries_classes.py</code> for the retrieving countries and disciplines that have more publications and journals
* <code>classes_description.ipynb</code> a Jupyter to describe classes attributes and methods so it is easer for my teammates to evaluate the structure
* <code>run_workflow.py</code> the python file containing the <code>def main()</code> that initializes classes instances to run the whole workflow.

16/05 we met to discuss next steps, and decided I will start writing the Article while Ali and Seba will do the visualizations

Yesterday I finished writing the code to retrieve missing countries from DOAJ and fixed some issues with cells that had more than one country in it. 

Today I started working on the article, read the paper <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3178846/">how to write your first research article</a> and designed an outline to discuss with the group.

## 19/05 - 21/05

With the group we further revised the software architecture and came up with the final classes structure.

## Classes description

### class Processor

<b>Attributes</b><br>
<code>batch_size</code> the size of the batches to process (int)<br>
<code>max_workers</code> the number of cpu workers to split the parallel processing<br>


### class Playarists Processor(Processor)

<b>Attributes</b><br>
<code>meta_path</code> the path to the OpenCitations Meta dump (needs to be unzipped)<br>
<code>erih_path</code> the path to the ERIH-PLUS dataset that is automatically initialized as a dataframe<br>
<code>doaj_path</code> the path to the DOAJ dataset that is automatically initialized as a dataframe<br>


<b>Methods</b><br>
<code>get_erih_df()</code> the method to return the ERIH-PLUS DataFrame<br>
<code>get_doaj_df()</code> the method to return the DOAJ DataFrame<br>
<code>process_files</code> returns a DataFrame with OpenCitations Meta SSH Journals included in ERIH-PLUS dataset and their Open_Access information (OC_omid, issn, EP_id, Publications_in_venue, Open Access)<br>

<hr>

### class ResultsProcessor

<b>Attributes</b><br>
<code>meta_coverage</code> a <span style= "color: green;">PlayaristsProcessor</span> object initialized as the result of the <span style= "color: blue;">process_files()</span> method<br>
<code>erih_df</code> the ERIH_PLUS DataFrame initialized by calling the <span style= "color: blue;">get_erih_df()</span> on the PlayaristsProcessor object<br>
<code>doaj_df</code> the DOAJ DataFrame initialized by calling the <span style= "color: blue;">get_doaj_df()</span> on the PlayaristsProcessor object<br>


<hr>

### class CountriesProcessor(ResultsProcessor)

<b>Attributes</b><br>
<code>doaj_df</code> a DOAJ sub-DataFrame with only issn and country columns


<b>Methods</b><br>
<code>create_countries_dict()</code> a method that returns a dictionary with Countries as keys and a list of each country's venue as value
<hr>

### class DisciplinesProcessor(ResultsProcessor)

<b>Methods</b><br>
<code>create_disciplines_dict()</code> a method that returns a dictionary with Disciplines as keys and a list of each discipline's venue as value

<hr>

### class CountsProcessor(ResultsProcessor)

<b>Attributes</b><br>
<code>export_path</code> the export path of the result file<br>


<b>Methods</b><br>
<code>counts()</code> takes in input a <span style= "color: green;">DisciplinesProcessor</span> or a <span style= "color: green;">CountriesProcessor</span>and a label (str) and returns a DataFrame with Label column (either Countries or Disciplines according to the dictionary in input), the count of venues and the count of Publications.


There have been minor changes after this, Sebastiano took care of them

## 22/05 - 23/05 and 27/05 - 28/05

I expanded the chosen bibliography to include some more recent studies and took some time to read these articles.

I wrote and edited the research article: Introduction, detailed Methodology ( starting from datasets description and the general process drafted by Sebastiano), Results interpretation and comparison with other studies + edited Discussion section starting from Ali's draft.

## 4/08 

Yesterday I had a look at the updates and adjustments made by Ali and Seba and envisioned a way to reply to the first three question in the issue.

Today I started writing the code to plot more data for meaningful comparison between US and UK Journals

#### Compare US and UK research question draft

* How many journals per country? US = 1161, UK = 1350


<b>Analysis</b>

United Kningdom has more journals but less publications, whereas US has less journal but more publication. Meaning that jpurnals in the us publish more compared to journals in the uk which appear to be less prolific.

* What kind of journals are they? what disciplines?
* What disciplines have more publications?
* What disciplines are published by more journals?

--> double barchart for counting the number of publication per discipline of only US and UK.
--> could do the same for journals

* What journals have more publications 

--> plot scatterplot that places each journal according to number of disciplines AND number of publications, so we can see if there's a correlation between the two





## 5/08

- I finished writing the code to process US and UK data
- made all the visualizations for the comparison between UK and US Journals, both html responsive and png
 
These are some notes for what I will need to write in the article:

big outlier in uk data:
- Nature with 207581 publications and 11 disciplines (megajournal):<br>
"Anthropology, Archaeology, Demography, Economics, Environmental Studies, Human Geography and Urban Studies, Interdisciplinary research in the Social Sciences, Media Studies and Communication, Science and Technology Studies, Social Statistics and Informatics, Sociology"

- every other journal is below 15000 publications

______________

In the us,
there are 4 mega journals with more than 100000 publications:
- PLOS ONE with 268538 publications and p Disciplines "Anthropology, Archaeology, Demography, Environmental Studies, History & Philosophy of Science, Human Geography and Urban Studies, Interdisciplinary research in the Social Sciences, Psychology, Sociology"
- Proceedings of the National Academy of Sciences of the United States of America, anthropology, 148780
- Science (EP_ID 341841), with 145090 and just Psychology
- Journal of the Acoustical Society of America, with 57919 and 2 disciplines, Linguistics, Musicology

________________

This will be useful also for addressing the question about how mega journals impact the results.

From plotting the correlation between number of publications per journal and number of disciplines per journal it is clearly observable 
and inverse trend, to explore more in depth.

<b>to do:</b>
- write interpretation
- integrate new code in software
- update results dataset
